src/pages/category/Category.js

```diff
@@ -1,33 +1,116 @@
 import { connect } from 'react-redux';
-import { useEffect } from 'react';
-import { setCategoryAsync, setCategory } from '../../store/categories';
+import { useEffect, useState } from 'react';
+import {
+  setCategoryAsync,
+  updateCategoryAsync,
+  createCategoryAsync,
+} from '../../store/categories';
+import { setProductsAsync } from '../../store/products';
+import { ListFilter } from '../../components/listFilter';
+import { ListEdit } from '../../components/listEdit';
+import { EditableField } from '../../components/editableField';
 import './category.scss';
 
-export const CategoryComponent = ({ match, dispatch, category }) => {
-  useEffect(() => {
-    dispatch(setCategoryAsync(match.params.id));
+export const CategoryComponent = ({
+  match, dispatch, category, products, user, prodAdded, history
+}) => {
+  const [categoryState, setCategoryState] = useState({ products: [] });
+  const newCategory = match.params.id === 'new';
 
-    return () => dispatch(setCategory({}));
+  useEffect(() => {
+    !newCategory && dispatch(setCategoryAsync(match.params.id));
+    dispatch(setProductsAsync());
   }, []);
 
+  const addProduct = (idProduct) => {
+    const product = products.find(item => item.id === idProduct);
+
+    if (newCategory) {
+      const arr = categoryState.products;
+      arr.push({ id: product.id, title: product.title });
+
+      setCategoryState({ ...categoryState, products: arr });
```

Comment:
just use `{ ...categoryState }` here, it already has `products`. Actually you don't need `arr` constant